src/code/z_demo.c

```diff
@@ -391,176 +443,189 @@ void func_80064824(PlayState* play, CutsceneContext* csCtx, CsCmdBase* cmd) {
+
+        case CS_MISC_SET_FLAG_4:
+            CutsceneFlags_Set(play, 4);
             break;
-        case 32:
-            if (sp3F != 0) {
+
+        case CS_MISC_SANDSTORM:
+            if (isFirstFrame) {
                 play->envCtx.sandstormState = SANDSTORM_FILL;
             }
+
             func_800788CC(NA_SE_EV_SAND_STORM - SFX_FLAG);
             break;
-        case 33:
+
+        case CS_MISC_SUNSSONG_START:
             gSaveContext.sunsSongState = SUNSSONG_START;
             break;
-        case 34:
+
+        case CS_MISC_FREEZE_TIME:
             if (IS_DAY) {
                 gSaveContext.dayTime -= gTimeSpeed;
             } else {
                 gSaveContext.dayTime -= gTimeSpeed * 2;
             }
             break;
-        case 35:
-            AudioOcarina_PlayLongScarecrowAfterCredits();
-            csCtx->frames = cmd->startFrame - 1;
+
+        case CS_MISC_LONG_SCARECROW_SONG:
+            AudioOcarina_PlayLongScarecrowSong();
+            csCtx->curFrame = cmd->startFrame - 1;
             break;
     }
 }
 
-// Command 4: Set Environment Lighting
-void Cutscene_Command_SetLighting(PlayState* play, CutsceneContext* csCtx, CsCmdEnvLighting* cmd) {
-    if (csCtx->frames == cmd->startFrame) {
-        play->envCtx.lightSettingOverride = cmd->setting - 1;
+void CutsceneCmd_SetLightSetting(PlayState* play, CutsceneContext* csCtx, CsCmdLightSetting* cmd) {
+    if (csCtx->curFrame == cmd->frame) {
+        play->envCtx.lightSettingOverride = cmd->settingPlusOne - 1;
         play->envCtx.lightBlend = 1.0f;
     }
 }
 
-// Command 0x56: Play Background Music
-void Cutscene_Command_PlayBGM(PlayState* play, CutsceneContext* csCtx, CsCmdMusicChange* cmd) {
-    if (csCtx->frames == cmd->startFrame) {
-        func_800F595C(cmd->sequence - 1);
+void CutsceneCmd_StartSequence(PlayState* play, CutsceneContext* csCtx, CsCmdStartSeq* cmd) {
+    if (csCtx->curFrame == cmd->frame) {
+        func_800F595C(cmd->seqIdPlusOne - 1);
     }
 }
 
-// Command 0x57: Stop Background Music
-void Cutscene_Command_StopBGM(PlayState* play, CutsceneContext* csCtx, CsCmdMusicChange* cmd) {
-    if (csCtx->frames == cmd->startFrame) {
-        func_800F59E8(cmd->sequence - 1);
+void CutsceneCmd_StopSequence(PlayState* play, CutsceneContext* csCtx, CsCmdStopSeq* cmd) {
+    if (csCtx->curFrame == cmd->frame) {
+        func_800F59E8(cmd->seqIdPlusOne - 1);
     }
 }
 
-// Command 0x7C: Fade Background Music over duration
-void Cutscene_Command_FadeBGM(PlayState* play, CutsceneContext* csCtx, CsCmdMusicFade* cmd) {
-    u8 var1;
+void CutsceneCmd_FadeSequence(PlayState* play, CutsceneContext* csCtx, CsCmdFadeSeq* cmd) {
+    u8 duration;
 
-    if ((csCtx->frames == cmd->startFrame) && (csCtx->frames < cmd->endFrame)) {
-        var1 = cmd->endFrame - cmd->startFrame;
+    if ((csCtx->curFrame == cmd->startFrame) && (csCtx->curFrame < cmd->endFrame)) {
+        duration = cmd->endFrame - cmd->startFrame;
```

Comment:
#1234 names this fadeTimer for now but the macro's arg name `fadeOutDuration` is probably better
https://github.com/zeldaret/oot/blob/170ff645c6c9b4fca9c0e2f11bba55c4a012c978/src/code/z_demo.c#L455-L458
https://github.com/zeldaret/oot/blob/170ff645c6c9b4fca9c0e2f11bba55c4a012c978/include/seqcmd.h#L87

-------

btw maybe the cutscene command should be renamed to StopSequence or FadeOutSequence, it wasn't clear to me that this was a "stop sequence" before I looked at seqcmd

but `CutsceneCmd_StopSequence` is something else already (StopSequenceNow), and idk if the difference between `CutsceneCmd_StopSequence` taking the player by seqId from seqFlags and `CutsceneCmd_FadeSequence` taking the player from the cs cmd argument is super significant

anyway probably revisit these names once both PRs are in I guess

Reply:
id say "stop" should still be given to the command that currently has it, and the fade one can just be clarified as "fade out"